### DISCUSSION TITLE
Set display while still transmitting data

### DISCUSSION THREAD

**Discussion by davystrong** [pvvx/ATC_MiThermometer#555] [Q&A]
Hello! I am trying to set up a thermometer display (MHO C401 clone) to show temperature from a different sensor. I'm using Home Assistant to do this, but my issue is not directly related to that. This uses the BTHome protocol to receive sensor data, and worked as expected.

With some help from one of the issues, I managed to get the display updating correctly to the _other value_ by sending 0x22 commands with a validity of 65535, which keeps that value permanently on the display, as intended.

However, I can't work out how best to make the two work together. When I am connected and writing to the display, my BTHome sensor values stop updating. Ideally, I would rather not have to stop using BTHome (and the linked Home Assistant extension) as getting notifications is better than polling.

Can anyone give me any suggestions? I believe the problem is that I want BTHome broadcasts while in "Connected" mode. What is the simplest way of achieving something similar?

**Comment by pvvx** [comment]
Simultaneous BLE connection and broadcasting of advertisements increases battery consumption. Therefore, this feature is not used.

During connection, the thermometer can transmit measurements at the interval set in the settings. To do this, you need to enable Notification.

Options for enabling automatic transmission of measurements:
1. There is an option in the configuration - when connecting and turning on "Notify", transmit measurements.
2. When connecting, send the command "33FF".  Only applies to the current connection.
    Commad "33xx" :  
"0x33" - Start/Stop notify measures in connection mode, 
"xx" - "0x01".. "0xFE" - number of transfers,  "0xff" - infinite, until the end of the connection, "0x00" - disable transfer. 

Sample: 
https://pvvx.github.io/ATC_MiThermometer/GraphAtc2.html 

There is another standard BLE option for transmitting measurements during connection.
When connecting, enable "Notify" for the temperature and humidity UUID. See BLE standards for a description.

Sample: 
https://pvvx.github.io/ATC_MiThermometer/GraphAtc1.html
https://pvvx.github.io/ATC_MiThermometer/GraphAtc.html

**Comment by davystrong** [comment]
Hello Victor, thank you SO much for your explanation! I don't know how you manage to be so patient and helpful with all the random questions!

I don't see that I'll be able to use existing HA extensions to do this then. However, that second option looks particularly interesting, so I'll look into writing something myself to handle that. Thank you again!
